Source/DafnyLanguageServer/Language/Symbols/SymbolTableFactory.cs

```diff
@@ -156,6 +157,12 @@ private class DesignatorVisitor : SyntaxTreeVisitor {
         RegisterDesignator(currentScope, frameExpression, frameExpression.tok, frameExpression.FieldName);
       }
 
+      public override void Visit(IdentifierExpr identifierExpression) {
+        cancellationToken.ThrowIfCancellationRequested();
+        RegisterDesignator(currentScope, identifierExpression, identifierExpression.tok, identifierExpression.Name);
```

Comment:
This may be the reason for the duplicated entries at a single location. It may even interfere with other designator registrations besides `Formal`.

Reply:
Oh good catch ! That's an insight.

Reply:
So the real answer wasn't related to that. A formal was added twice because the syntax tree visitor was visiting an auto-generated `decrease` expression, which turns out takes the same token as the formal. So I removed the visit of auto-generated expressions during syntax tree visit to ensure we don't add the same thing twice.

Reply:
> So I removed the visit of auto-generated expressions during syntax tree visit to ensure we don't add the same thing twice.

Good catch. I remember that I had problems with the auto-generated expressions in the past for the same reason. At this time, I don't think that the language server requires visiting auto-generated expressions, so that should be fine in the base visitor implementation. However, this should be documented (e.g., as a `remark` entry in the XML-Doc of the class). Later, we could make it configurable through a constructor parameter if there's a need.